File: Polimorfismo/fraccion.py
class Fraccion:
    def __init__(self,a,b):
        self.__numerador = a
        self.__denominador = b
    '''
      c = a + b
      c.numerador = a.numerador * b.denominador +
      a.denominador * b.numerador
      c.denominador = a.denominador * b.denominador
    '''
    def __add__(self, o):
        a = self.__numerador * o.__denominador + \
           self.__denominador * o.__numerador
        b = self.__denominador * o.__denominador
        return Fraccion(a,b)
    '''
      c = a - b
      c.numerador = a.numerador * b.denominador -
        a.denominador * b.numerador
      c.denominador = a.denominador * b.denominador
    '''
    def __sub__(self, o):
        c = Fraccion(0,1)
        c.__numerador = self.__numerador * o.__denominador - \
           self.__denominador * o.__numerador
        c.__denominador = self.__denominador * o.__denominador
        return c
    def __str__(self):
        return "{}/{}". \
          format(self.__numerador,self.__denominador)
    
    

File: Polimorfismo/Fraccion.java
package poo;

public class Fraccion {
	// Atributos
	private int numerador;
	private int denominador;
	
	// Metodos
	public Fraccion(int a, int b){
		this.numerador = a;
		this.denominador = b;
	}
	/*
	 c = a.suma(b) =>  c = a + b
	 c.numerador = a.numerador * b.denominador +
	            a.denominador * b.numerador
	 c.denominador = a.denominador * b.denominador
	*/
	
	public Fraccion suma(Fraccion o){
		int a = this.numerador * o.denominador +
				this.denominador * o.numerador;
		int b = this.denominador * o.denominador;
		return new Fraccion(a,b);
	}
	/*
	 c = a.resta(b) =>  c = a - b
	 c.numerador = a.numerador * b.denominador -
	            a.denominador * b.numerador
	 c.denominador = a.denominador * b.denominador
	*/
	public Fraccion resta(Fraccion o){
		Fraccion c = new Fraccion(0, 1);
		c.numerador = this.numerador * o.denominador -
				this.denominador * o.numerador;
		c.denominador = this.denominador * o.denominador;
		return c;
	}
	
	public String toString(){
		return String.format("%d/%d", this.numerador,this.denominador);
	}
}
